File: NthFibonacciNumber.py
def nth_fibonacci(n):
    if n <= 1:
        return n

    curr = 0   # Biến lưu trữ số Fibonacci hiện tại
    prev1 = 1  # Biến lưu trữ số Fibonacci trước đó (F(n-1))
    prev2 = 0  # Biến lưu trữ số Fibonacci trước nữa (F(n-2))

    for i in range(2, n + 1):
        # Tính giá trị Fibonacci hiện tại
        curr = prev1 + prev2
        # Cập nhật prev2 thành prev1 (F(n-2) trở thành F(n-1) cho vòng lặp tiếp theo)
        prev2 = prev1
        # Cập nhật prev1 thành curr (F(n-1) trở thành F(n) cho vòng lặp tiếp theo)
        prev1 = curr

    # Sau khi vòng lặp kết thúc, curr sẽ chứa số Fibonacci thứ n
    return curr

n = int(input("Nhập số n: "))

result = nth_fibonacci(n)
print(f"Số Fibonacci thứ {n} là: {result}")

File: NthFibonacciNumber.c
#include <stdio.h>
int nthFibonacci(int n) {
    if (n < 0) return -1; // Trả về lỗi nếu n không hợp lệ (n phải >= 0)
    if (n == 1 || n == 0) return n;

    int previous = 0, current = 1; // Đặt giá trị Fibonacci ban đầu
    for (int i = 2; i <= n; i++) {
        int next = previous + current; // Tính Fibonacci tiếp theo
        previous = current;            // Cập nhật giá trị trước đó
        current = next;                // Cập nhật giá trị hiện tại
    }
    return current;  // Trả về Fibonacci thứ n
}
int main() {
    int n;
    printf("Nhập chỉ số n: ");
    scanf("%d", &n);
    int result = nthFibonacci(n);
    if (result == -1) {
        printf("Chỉ số n không hợp lệ. Vui lòng nhập n >= 0.\n");
    } else {
        printf("Fibonacci thứ %d là: %d\n", n, result);
    }

    return 0;
}
